src/ome_zarr_models/_rfc5_transforms/multiscales.py

```diff
@@ -0,0 +1,247 @@
+          the first transformation is a scale and the second a translation
+        """
+
+        class Transforms(BaseModel):
+            transforms: tuple[CoordinateTransformationType, ...]
+
+        transforms = Transforms(transforms=transforms_obj).transforms
+        check_length(transforms, valid_lengths=[1], variable_name="transforms")
+
+        transform = transforms[0]
+        if transform.type == "sequence":
+            check_length(
+                transform.transformations,
+                valid_lengths=[2],
+                variable_name="transform.transforms (i.e. transformations composing "
+                "the sequence)",
+            )
+            first, second = transform.transformations
+            if first.type != "scale":
+                msg = (
+                    "When the first (and only) element in `coordinateTransformations`"
+                    " is a `Sequence`, the first element must be a `Scale` transform. "
+                    f"Got {first} instead."
+                )
+                raise ValueError(msg)
+            if second.type != "translation":
+                msg = (
+                    "When the first (and only) element in `coordinateTransformations`"
+                    " is a `Sequence`, the second element must be a `Translation` "
+                    f"transform. Got {second} instead."
+                )
+                raise ValueError(msg)
+        elif transform.type != "scale":
+            msg = (
+                "The first transformation in `coordinateTransformations` "
+                "must either be a `Scale` transform or a `Sequence` transform. "
+                f"Got {transform} instead."
+            )
+            raise ValueError(msg)
+
+        return transforms_obj
+
+    @field_validator("coordinateTransformations", mode="after")
+    @classmethod
+    def _ensure_transform_dimensionality(
+        cls,
+        transforms: tuple[CoordinateTransformationType, ...],
+    ) -> tuple[CoordinateTransformationType, ...]:
+        """
+        Ensures that the dimensionality of the scale and translation (when both present)
+        match
+        """
+        maybe_sequence = transforms[0]
+        if maybe_sequence.type == "sequence":
+            first, second = maybe_sequence.transformations
+            if (
+                isinstance(first, Scale)
+                and isinstance(second, Translation)
+                and len(first.scale) != len(second.translation)
+            ):
+                raise ValueError(
+                    "The length of the scale and translation vectors must be the same."
+                    f"Got {len(first.scale)} and {len(second.translation)}."
+                )
+        return transforms
+
+
+class Multiscale(BaseAttrs):
+    """
+    An element of multiscales metadata.
+    """
+
+    coordinateSystems: tuple[CoordinateSystem, ...] = Field(..., min_length=1)
```

Comment:
does the spec allow multiple coordinate systems with the same `name` attribute?

Reply:
i checked, and the rfc5 states that the `name` attributes must be unique across elements of `coordinateSystems`. this means that a tuple is actually unnecessary, and this should be a dict instead. but that requires changing rfc5.

Reply:
another issue with the spec: it states that the name of an array can be used to denote that array's coordinate system. but it's unclear what should happen if the name of the array is already used by an element of `coordinateSystems`.

Reply:
I agree with both and thanks for pointing this out in Zulip. Could you please open a GitHub issue for both? I am afraid that otherwise these important points may get lost.

Reply:
Actually, I created such issue: https://github.com/ome-zarr-models/ome-zarr-models-py/issues/189.

Reply:
I would resolve this conversation since the code todo items can be discussed in the linked issue and tackled in a follow up PR.